panel/chat/chatarea_input.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING, ClassVar, Mapping, Optional, Type,
+)
+
+import param
+
+from ..models.chatarea_input import (
+    ChatAreaInput as _bkChatAreaInput, ChatMessageEvent,
+)
+from ..widgets import TextAreaInput as _PnTextAreaInput
+
+if TYPE_CHECKING:
+    from bokeh.document import Document
+    from pyviz_comms import Comm
+
+from bokeh.model import Model
+
+
+class ChatAreaInput(_PnTextAreaInput):
+    """
+    The `ChatAreaInput` allows entering any multiline string using a text input
+    box, with the ability to click enter to submit the message.
+
+    Unlike TextAreaInput, the `ChatAreaInput` defaults to auto_grow=True and
+    max_rows=10, and the value is not synced to the server until the enter key
+    is pressed so key on `value_input` if you need to access the existing value.
+
+    Lines are joined with the newline character `\n`.
+
+    Reference: https://panel.holoviz.org/reference/widgets/ChatAreaInput.html
+    :Example:
+
+    >>> ChatAreaInput(max_rows=10)
+    """
+
+    auto_grow = param.Boolean(default=True, doc="""
+        Whether the text area should automatically grow vertically to
+        accommodate the current text.""")
+
+    max_rows = param.Integer(default=10, doc="""
+        When combined with auto_grow this determines the maximum number
+        of rows the input area can grow.""")
+
+    resizable = param.ObjectSelector(
+        default="height",
+        objects=["both", "width", "height", False],
+        doc="""
+        Whether the layout is interactively resizable,
+        and if so in which dimensions: `width`, `height`, or `both`.
+        Can only be set during initialization.""",
+    )
+
+    _widget_type: ClassVar[Type[Model]] = _bkChatAreaInput
+
+    _rename: ClassVar[Mapping[str, str | None]] = {
+        "value": None, **_PnTextAreaInput._rename,
```

Comment:
If we want to be able to provide an initial message (which I'm not sure about), we'd have to add:

```python
def _get_properties(self, doc: Document) -> Dict[str, Any]:
    props = super()._get_properties(doc)
    props['value'] = self.value
    return props
```